src/AasxServerDB/EntityFW.cs

```diff
@@ -0,0 +1,323 @@
+            if (_con == null)
+            {
+                options.UseNpgsql("");
+            }
+            else
+            {
+                var connectionString = _con["DatabaseConnection:ConnectionString"];
+                if (connectionString != null && connectionString.Contains("$DATAPATH"))
+                    connectionString = connectionString.Replace("$DATAPATH", _dataPath);
+                options.UseNpgsql(connectionString);
+            }
+        }
+    }
+
+    public class AASXSet
+    {
+        public int Id { get; set; }
+        public string AASX { get; set; }
+
+        public virtual ICollection<AASSet> AASSets { get; } = new List<AASSet>();
+        public virtual ICollection<SMSet> SMSets { get; } = new List<SMSet>();
+    }
+
+    public class AASSet
+    {
+        public int Id { get; set; }
+
+        [ForeignKey("AASXSet")]
+        public int AASXId { get; set; }
+        public virtual AASXSet AASXSet { get; set; }
+
+        public string? Identifier { get; set; }
+        public string? IdShort { get; set; }
+        public string? AssetKind { get; set; }
+        public string? GlobalAssetId { get; set; }
+
+        public virtual ICollection<SMSet> SMSets { get; } = new List<SMSet>();
+    }
+
+    public class SMSet
+    {
+        public int Id { get; set; }
+
+        [ForeignKey("AASXSet")]
+        public int AASXId { get; set; }
+        public virtual AASXSet AASXSet { get; set; }
+
+        [ForeignKey("AASSet")]
+        public int AASId { get; set; }
+        public virtual AASSet AASSet { get; set; }
+
+        public string? SemanticId { get; set; }
+        public string? Identifier { get; set; }
+        public string? IdShort { get; set; }
+
+        public virtual ICollection<SMESet> SMESets { get; } = new List<SMESet>();
+    }
+
+    public class SMESet
+    {
+        public int Id { get; set; }
+
+        [ForeignKey("SMSet")]
+        public int SMId { get; set; }
+        public virtual SMSet SMSet { get; set; }
+
+        public int? ParentSMEId { get; set; }
+        public virtual SMESet? ParentSME { get; set; }
+
+        public string? SMEType { get; set; }
+        public string? ValueType { get; set; }
+        public string? SemanticId { get; set; }
+        public string? IdShort { get; set; }
+
+        public virtual ICollection<IValueSet> IValueSets { get; } = new List<IValueSet>();
+        public virtual ICollection<DValueSet> DValueSets { get; } = new List<DValueSet>();
+        public virtual ICollection<SValueSet> SValueSets { get; } = new List<SValueSet>();
+
+        public string getValue()
+        {
+            using (AasContext db = new AasContext())
+            {
+                switch (ValueType)
+                {
+                    case "S":
+                        var ls = db.SValueSets.Where(s => s.SMEId == Id).Select(s => s.Value).ToList();
+                        if (ls.Count != 0)
+                            return ls.First() + "";

```

Comment:
Don't use `+ ""` to make it a string. I don't know who started that here, but either it is a string already or you should use `ToString()`